test/extensions/filters/http/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,135 @@
+#include "test/extensions/filters/http/common/fuzz/uber_filter.h"
+
+#include "common/config/utility.h"
+#include "common/config/version_converter.h"
+#include "common/http/message_impl.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+
+UberFilterFuzzer::UberFilterFuzzer() {
+  // Need to set for both a decoder filter and an encoder/decoder filter.
+  ON_CALL(filter_callback_, addStreamDecoderFilter(_))
+      .WillByDefault(Invoke([&](std::shared_ptr<Envoy::Http::StreamDecoderFilter> filter) -> void {
+        filter_ = filter;
+        filter_->setDecoderFilterCallbacks(callbacks_);
+      }));
+  ON_CALL(filter_callback_, addStreamFilter(_))
+      .WillByDefault(Invoke([&](std::shared_ptr<Envoy::Http::StreamDecoderFilter> filter) -> void {
+        filter_ = filter;
+        filter_->setDecoderFilterCallbacks(callbacks_);
+      }));
+  // Set expectations for particular filters that may get fuzzed.
+  setExpectations();
+}
+
+void UberFilterFuzzer::setExpectations() {
+  prepareExtAuthz();
+  prepareCache();
+  prepareTap();
+}
+
+void UberFilterFuzzer::prepareExtAuthz() {
```

Comment:
Yeah, let's move these to per-filter as well.

Reply:
done - but let me know if you dislike how i merged them to the same method. i wanted to avoid passing the mocks as params, but can change that quickly.